tests/dataset_test.py

```diff
@@ -765,3 +765,136 @@ def test_derive_removes_attachments(initial, attachments):
     initial.attachments = attachments
     derived = initial.derive()
     assert derived.attachments == []
+
+
+@pytest.fixture(params=[DatasetType.RAW, DatasetType.DERIVED])
+def my_type(request):
+    return request.param
+
+
+@pytest.fixture
+def invalid_field_example(my_type):
+    if my_type == DatasetType.DERIVED:
+        return "data_format", "sth_not_None"
+    elif my_type == DatasetType.RAW:
+        return "job_log_data", "sth_not_None"
+    else:
+        raise ValueError(my_type, " is not valid DatasetType.")
+
+
+def test_dataset_dict_like_keys_per_type(my_type):
```

Comment:
Can you use the datasets strategy to generate input datasets? The strategy always makes valid datasets if you set `for_upload=True`.